js/data/array_query.js

```diff
@@ -362,7 +380,8 @@ const FilterIterator = WrappedIterator.inherit({
 
     ctor: function(iter, criteria) {
         this.callBase(iter);
-        this.criteria = compileCriteria(criteria);
+        this.compareOptions = iter.compareOptions;
```

Comment:
Question.

Do we need the field ```compareOptions``` in this class?
I didn't see any usage of the ```this.compareOptions```
But maybe I missed something :)

Reply:
we need it. In other places new iterators wrap others, so we need to store compareOptions in those chains.